### PR TITLE
fix: pathlib.WindowsPath converts Win paths to lowercase

### PR DESCRIPTION
**Pull request by kutny**
# Pull Request Check List

This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://poetry.eustace.io/docs/contributing/) at least once, it will save you unnecessary review cycles!

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

Hi,

Poetry uses WindowsPath which, on Windows, unexpectedly converts file system paths to lowercase. As a result, I am not able to correctly import classes from packages built by Poetry on Windows.

**real Windows path:**
src/DbxDeploy/Logger/MyLogger.py

**path in package build by Poetry:**
src/dbxdeploy/Logger/MyLogger.py

See also: https://stackoverflow.com/questions/21173979/how-do-i-get-path-to-python-script-with-proper-case

Thank you for considering my fix

### CODE REVIEW

**Comment by kutny** [comment]
Hi @sdispater 

I do not understand why the build is failing on files, that I haven't changed at all:

```
Files were modified by this hook. Additional output:

Skipped 7 files
Fixing /home/runner/work/poetry/poetry/tests/utils/fixtures/setups/pendulum/setup.py
Fixing /home/runner/work/poetry/poetry/tests/utils/fixtures/setups/ansible/setup.py
Fixing /home/runner/work/poetry/poetry/tests/utils/fixtures/setups/sqlalchemy/setup.py
```

https://github.com/sdispater/poetry/pull/1508/checks?check_run_id=286072237

Could you help, please?

**Comment by kutny** [comment]
The files are obviously not fixed by isort, so it might a bug that is caused by https://github.com/sdispater/poetry/commit/6b09639002e737e76a09eb973f0049262511d3f0

**Comment by kutny** [comment]
Thanks for fixing the isort bug @sdispater 

This PR is ready to be merged from my side. Is there anything else you want me to add to have it merged? It is a very simple change and without that, I am not able to build our company packages at all. Thank you

**Comment by kutny** [comment]
Hi @sdispater, any chance of having this merged soon? It is a pretty simple change, can I improve it anyhow? Thanks

File: poetry/masonry/utils/include.py [hunk 1]
```diff
@@ -25,7 +26,7 @@ def __init__(
         self._include = str(include)
         self._formats = formats
 
-        self._elements = sorted(list(self._base.glob(str(self._include))))
+        self._elements = sorted(list(self._glob_include()))
```
Comment (finswimmer) [inline poetry/masonry/utils/include.py]: What about

```
self._elements = sorted(list(glob(Path(self._base, self._include), recursive=True)))
```
?

AFAIR `pathlib`'s `glob` doesn't  return the correct case on windows file system. So replacing it in this line with the "normal" glob should be enough, shouldn't it?

Comment (kutny) [reply poetry/masonry/utils/include.py]: @finswimmer Your solution returns the "expected string or bytes-like object" error, not sure why...

Comment (kutny) [reply poetry/masonry/utils/include.py]: I think that glob() expects the first argument to be string, not Path. If I use str(Path(..., I get another error: 'str' object has no attribute 'is_dir'

**Comment by abn** [comment]
@kutny is this somehting you still want to look at? If so this might need to be reworked; also the relevant logic is now in poetry-core.

**Comment by neersighted** [comment]
Closing this for now for staleness. PRs against poetry-core (as well as detailed reproductions) are welcomed!

**Pull request closed without merging**